src/tests/testIpAddress.cc

```diff
@@ -532,20 +532,21 @@ TestIpAddress::testtoUrl_fromSockAddr()
     anIPA.toUrl(buf,MAX_IPSTRLEN);
     CPPUNIT_ASSERT( memcmp("192.168.100.12:80", buf, 17) == 0 );
 
-    /* test output when constructed from in6_addr with IPv6 */
-    struct sockaddr_in6 ip6val;
-
-    ip6val.sin6_addr.s6_addr32[0] = htonl(0xC0A8640C);
-    ip6val.sin6_addr.s6_addr32[1] = htonl(0xFFFFFFFF);
-    ip6val.sin6_addr.s6_addr32[2] = htonl(0xFFFFFFFF);
-    ip6val.sin6_addr.s6_addr32[3] = htonl(0xFFFFFFFF);
-    ip6val.sin6_port = htons(80);
-    ip6val.sin6_family = AF_INET6;
+    /* test output when constructed from sockaddr_storage with IPv6 */
+    struct sockaddr_storage ss;
+    auto *ip6val = reinterpret_cast<struct sockaddr_in6*>(&ss);
+
+    ip6val->sin6_addr.s6_addr32[0] = htonl(0xC0A8640C);
+    ip6val->sin6_addr.s6_addr32[1] = htonl(0xFFFFFFFF);
+    ip6val->sin6_addr.s6_addr32[2] = htonl(0xFFFFFFFF);
+    ip6val->sin6_addr.s6_addr32[3] = htonl(0xFFFFFFFF);
+    ip6val->sin6_port = htons(80);
+    ip6val->sin6_family = AF_INET6;
 #if HAVE_SIN6_LEN_IN_SAI
-    ip6val.sin6_len = sizeof(struct sockaddr_in6);
+    ip6val->sin6_len = sizeof(struct sockaddr_in6);
 #endif
 
-    Ip::Address bnIPA(ip6val);
+    Ip::Address bnIPA(ss);
```

Comment:
If possible:

```suggestion
    const Ip::Address bnIPA(reinterpret_cast<struct sockaddr_storage&>(ip6val));
```

Reply:
I am reluctant to do this one. While it matches up with the earlier IPv4 constructor test case, the intention of this case is to verify that the `sockaddr_storage` content is correctly interpreted by `Ip::Address` when being set via an `ipv6val` pointer/reference.

Reply:
I will rephrase to avoid contaminating this trivial change request with complex changes from another one:

```suggestion
    const Ip::Address bnIPA(ss);
```